### PR TITLE
Snap 293

### PR DESCRIPTION
**Pull request by nthanvi**
changes required to support the dynamic Jar loading  in the Snappy.

we have created our own implementation of Executor and for that  introduced new method in the base classes which are overwritten by our implementation.

Related PR -
https://github.com/SnappyDataInc/spark-jobserver/pull/1
https://github.com/SnappyDataInc/snappy-spark/pull/39
https://github.com/SnappyDataInc/snappy-store/pull/88
https://github.com/SnappyDataInc/snappydata/pull/310

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/CatalystSchemaConverter.scala [hunk 1]
```diff
@@ -566,7 +566,8 @@ private[parquet] object CatalystSchemaConverter {
   }
 
   // Returns the minimum number of bytes needed to store a decimal with a given `precision`.
-  val minBytesForPrecision = Array.tabulate[Int](DecimalType.MAX_PRECISION + 1)(computeMinBytesForPrecision)
+  val minBytesForPrecision =
+    Array.tabulate[Int](DecimalType.MAX_PRECISION + 1)(computeMinBytesForPrecision)
```
Comment (sumwale) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/CatalystSchemaConverter.scala]: avoid this change

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 2]
```diff
@@ -371,13 +371,20 @@ private[spark] class Executor(
...
+      parentLoader: ClassLoader, userClassPathFirst: Boolean): MutableURLClassLoader = {
     if (userClassPathFirst) {
-      new ChildFirstURLClassLoader(urls, currentLoader)
+      new ChildFirstURLClassLoader(urls, parentLoader)
     } else {
-      new MutableURLClassLoader(urls, currentLoader)
+      new MutableURLClassLoader(urls, parentLoader)
     }
   }
-
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala]: revert this

**Comment by sumwale** [comment]
looks good, some minor comments

**Comment by nthanvi** [comment]
already commited after merging to SNAP_293 after spark 2.0 merge

**Pull request closed without merging**